src/core/lib/channel/compress_filter.c

```diff
@@ -242,7 +242,8 @@ static void compress_start_transport_stream_op(grpc_exec_ctx *exec_ctx,
     process_send_initial_metadata(elem, op->send_initial_metadata);
   }
   if (op->send_message != NULL && !skip_compression(elem) &&
-      0 == (op->send_message->flags & GRPC_WRITE_NO_COMPRESS)) {
+      0 == (op->send_message->flags &
```

Comment:
While you are here, could you move the `0 == (op->send_message->flags & (GRPC_WRITE_NO_COMPRESS | GRPC_WRITE_INTERNAL_COMPRESS))` clause into `skip_compression`?